app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java

```diff
@@ -0,0 +1,20 @@
+package ru.yandex.practicum.contacts.presentation.base;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.DiffUtil;
+
+
+public class BaseListDiffCallback
+        <T extends ListDiffInterface<T>>
+        extends DiffUtil.ItemCallback<T> {
+
+    @Override
+    public boolean areItemsTheSame(@NonNull T oldItem, @NonNull T newItem) {
+        return oldItem.theSameAs(newItem);
+    }
+
+    @Override
+    public boolean areContentsTheSame(@NonNull T oldItem, @NonNull T newItem) {
+        return oldItem.theSameAs(newItem);
```

Comment:
```suggestion
        return oldItem.equals(newItem);
```
В колбэке `areContentsTheSame` необходимо сравнить содержимое объектов на равенство, поэтому здесь подойдёт функция `equals`